gcloud/datastore/query.py

```diff
@@ -463,6 +463,11 @@ def __iter__(self):
                 yield entity
             if not self._more_results:
                 break
+            num_results = len(self._page)
+            if self._limit is not None:
+                self._limit -= num_results
+            if self._offset is not None:
+                self._offset += num_results
```

Comment:
The offset will always get smaller. For example:

     query().offset(1500).limit(400)  # I want to skip 1500 results, then get the next 400.

Datastore will only skip 1000, so you'll get a result like:

   {
       end_cursor: XXX,
       num_skipped_results: 1000,
       results: [],
       more_results: NOT_FINISHED,
   }

The client should automatically continue the query for the user, so it should issue another query:

     query().offset(500).limit(400)

Datastore may not return the full limit, so you may get something like (100 results):

     
   {
       end_cursor: XXX,
       num_skipped_results: 500,
       results: [a, b, c, ....],  # 200 results
       more_results: NOT_FINISHED
   }

Then, the client should again issue another query for the user:

     query().limit(200)  # No offset since it has been satisfied.

Datastore then may return the rest of the dataset:

   {
       end_cursor: XXX,
       num_skipped_results: 0,
       results: [a, b, c, ....],  # 200 results
       more_results: MORE_RESULTS_AFTER_LIMIT
   }